core/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -1002,15 +1013,15 @@ public void parseXContent(QueryParseContext context, AggregatorParsers aggParser
                         scriptFields.add(new ScriptField(context));
                     }
                 } else if (context.getParseFieldMatcher().match(currentFieldName, INDICES_BOOST_FIELD)) {
-                    indexBoost = new ObjectFloatHashMap<>();
+                    indexBoost = new ArrayList<>();
```

Comment:
shall we also deprecate this old syntax in 5.x given that it doesn't support ordering, in favour of the new array syntax added below?